AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/ContextCompatAwareResHandler.java

```diff
@@ -66,14 +64,11 @@ private boolean hasTargetMethodInContext() {
 		return hasTargetMethod(context, androidRes.getResourceMethodName());
 	}
 
-	private void createCallWithIfGuard(EComponentHolder holder, JFieldRef ref, JBlock methodBody, JFieldRef idRef) {
+	private IJExpression createCallWithIfGuard(EComponentHolder holder, JFieldRef idRef) {
 		JVar resourcesRef = holder.getResourcesRef();
-		JConditional guardIf = methodBody._if(getClasses().BUILD_VERSION.staticRef("SDK_INT").gte(getClasses().BUILD_VERSION_CODES.staticRef(minSdkPlatformName)));
```

Comment:
Can't we move back to the normal if instead of the inline if here? (as no we have the temporary variable)